impl/src/main/java/org/jboss/weld/event/ObserverNotifier.java

```diff
@@ -200,9 +201,15 @@ public void fireEvent(Object event, Resolvable resolvable) {
     protected Resolvable buildEventResolvable(Type eventType, Set<Annotation> qualifiers) {
         // We can always cache as this is only ever called by Weld where we avoid non-static inner classes for annotation literals
         Set<Type> typeClosure = sharedObjectCache.getTypeClosureHolder(eventType).get();
-        return new ResolvableBuilder(resolver.getMetaAnnotationStore()).addTypes(typeClosure).addType(Object.class)
+        ResolvableBuilder resolvableBuilder = new ResolvableBuilder(resolver.getMetaAnnotationStore()).addTypes(typeClosure)
+                .addType(Object.class)
                 .addQualifiers(qualifiers)
-                .addQualifierUnchecked(QualifierInstance.ANY).create();
+                .addQualifierUnchecked(QualifierInstance.ANY);
+        if (qualifiers.isEmpty()) {
+            resolvableBuilder.addQualifier(Default.Literal.INSTANCE);
+        }
+        return resolvableBuilder.create();
+
     }
 
     protected Resolvable buildEventResolvable(Type eventType, Annotation... qualifiers) {
```

Comment:
I guess you also need to modify this method?

Reply:
Hm, I wonder why we don't just delegate the method via `buildEventResolvable(eventType, Set.of(qualifiers));`.
We should probably do that.

Reply:
That makes sense to me :-)

Reply:
Also, the reason why ... is probably to save a few allocations, I can't think of anything else.